.gitlab-ci.yml

```diff
@@ -1,48 +1,63 @@
 variables:
   GIT_SUBMODULE_STRATEGY: recursive
 
-cache:
-  key: ${CI_COMMIT_REF_SLUG}
-  paths:
-    - node_modules/
-    - dist/
-    - proto/
-    - docs/
-
 stages:
   - check
   - build
   - deploy
 
-before_script:
-  - node --version
-  - npm install
-
-prettier:
+lint:
+  image: registry.gitlab.com/matrixai/engineering/maintenance/gitlab-runner
   stage: check
-  image: node:12
   script:
-    - npm run lint
+    - nix-shell --packages nodejs --run "npm install"
```

Comment:
Can start moving towards using `nix-shell -I nixpkgs=./pkgs.nix` if you use the same style as typescript-demo-lib.

Can be done in the next PR.

Reply:
`pkgs.nix` has been changed to a function export and called in `shell.nix`.
CI/CD file has been changed to the syntax you mentioned.